dev/tests/js/jasmine/tests/app/code/Magento/Ui/base/js/grid/data-storage.test.js

```diff
@@ -22,7 +25,253 @@ define([
+                var ids = [1,2,3],
+                    type = typeof model.getIds(ids);
+
+                expect(type).toEqual('object');
+            });
+
+        });
+
+        describe('"getData" method', function () {
+
+            var model = new DataStorage({
+                dataScope: 'magento'
+            });
+
+            it('check for defined', function () {
+                expect(model.hasOwnProperty('getData')).toBeDefined();
+            });
+
+            it('check method type', function () {
+                expect(typeof model.getData).toEqual('function');
+            });
+
+            it('check returned value if method called with argument', function () {
+                var params = {
+                    namespace: 'magento',
+                    search: '',
+                    filters: {
+                        store_id: 0
+                    },
+                    sorting: {},
+                    paging: {}
+                };
+
+                expect(model.getData(params)).toBeDefined();
+            });
+
+            it('check returned type if method called with argument', function () {
+                var params = {
+                    namespace: 'magento',
+                    search: '',
+                    filters: {
+                        store_id: 0
+                    },
+                    sorting: {},
+                    paging: {}
+                },
+                    type = typeof model.getData(params);
+
+                expect(type).toEqual('object');
+            });
+
+            it('check "clearRequests" has been called', function () {
+                var params = {
+                    namespace: 'magento',
+                    search: '',
+                    filters: {
+                        store_id: 0
+                    },
+                    sorting: {},
+                    paging: {}
+                };
+
+                spyOn(model, 'clearRequests');
+                spyOn(model, 'hasScopeChanged').and.callFake(function () {
+                    return true;
+                });
+                model.getData(params);
+                expect(model.clearRequests).toHaveBeenCalled();
+            });
+
+            it('check "getRequest" has been called', function () {
+                var params = {
+                    namespace: 'magento',
+                    search: '',
+                    filters: {
+                        store_id: 0
+                    },
+                    sorting: {},
+                    paging: {}
+                };
+
+                spyOn(model, 'getRequest');
+                spyOn(model, 'hasScopeChanged').and.callFake(function () {
+                    return false;
+                });
+                model.getData(params);
+                expect(model.getRequest).toHaveBeenCalled();
+            });
+
+            it('Return "getRequestData" method', function () {
```

Comment:
One thing I think we should try to do is provide good test names. This may seem like a silly recommendation, but as a first-time contributor to the project, one of the first things I do to familiarize myself with the project is read the tests! While I understand what the intention behind this test is, it only became clear to me after reading the data-storage `getData` method.
I would suggest naming the test something like "it returns cached request  data if a cached request exists and no refresh option is provided". In this way, the name of the test alone is enough to explain to me some key functionality of this module.

Reply:
I'd agree that test names should be more detailed to differentiate test between each other, however, I wouldn't make them too long. I don't think test results can be accessed without reading the code (by name only).